stdlib/public/runtime/ProtocolConformance.cpp

```diff
@@ -142,11 +142,46 @@ const {
   case ProtocolConformanceReferenceKind::WitnessTableAccessor:
     return getWitnessTableAccessor()(type, nullptr, 0);
 
-  case ProtocolConformanceReferenceKind::ConditionalWitnessTableAccessor:
+  case ProtocolConformanceReferenceKind::ConditionalWitnessTableAccessor: {
     // FIXME: this needs to query the conditional requirements to form the
     // array of witness tables to pass along to the accessor.
+    const char *typeName = "<unknown type>";
+    switch (getTypeKind()) {
+    case TypeMetadataRecordKind::NonuniqueDirectType:
+    case TypeMetadataRecordKind::UniqueDirectType:
+      if (const auto *ntd = getDirectType()->getNominalTypeDescriptor())
+        typeName = ntd->Name.get();
+      break;
+
+    case TypeMetadataRecordKind::UniqueDirectClass:
+#if SWIFT_OBJC_INTEROP
+      typeName = class_getName(getDirectClass());
+#endif
+      break;
+
+    case TypeMetadataRecordKind::UniqueIndirectClass:
+#if SWIFT_OBJC_INTEROP
+      typeName = class_getName(*getIndirectClass());
+#endif
+      break;
+
+    case TypeMetadataRecordKind::UniqueNominalTypeDescriptor: {
+      SymbolInfo info;
+      if (lookupSymbol(getNominalTypeDescriptor(), &info))
+        typeName = info.symbolName;
+      break;
+    }
+
+    case TypeMetadataRecordKind::Universal:
+      break;
+    }
+    warning(/*flag=*/0,
+            "warning: Swift runtime does not yet support dynamically "
+            "querying conditional conformance ('%s': '%s')\n",
+            typeName, getProtocol()->Name);
```

Comment:
Do you have the original source type metadata pointer from the cast at this point? If so, you can get a presentable string by passing it to `swift_getTypeName`.

Reply:
Ah, I missed `swift_getTypeName()`. I'll try that.